src/envoy/mixer/utils.cc

```diff
@@ -54,8 +60,13 @@ bool GetSourceUser(const Network::Connection* connection, std::string* user) {
   if (connection) {
     Ssl::Connection* ssl = const_cast<Ssl::Connection*>(connection->ssl());
     if (ssl != nullptr) {
-      *user = ssl->uriSanPeerCertificate();
-      return true;
+      std::string result = ssl->uriSanPeerCertificate();
+      if (result.length() >= kSPIFFEPrefix.length() &&
+          result.compare(0, kSPIFFEPrefix.length(), kSPIFFEPrefix) == 0) {
+        // Strip out the prefix "spiffe://" in the identity.
+        *user = result.substr(kSPIFFEPrefix.size());
+        return true;
+      }
```

Comment:
what about else? the one doesn't match the prefix?

Reply:
if not, we don't need to do anything. So far, all id will have this prefix.

Reply:
I think we still want to get source.user even it doesn't match spiff prefix.

Reply:
what format will it look like?   will it have similar format of xyz://  If so,  we want to strip it too.

Reply:
Good point, as mentioned, so far, all Ids have prefix "spiffe://", i don't think we have plan to support others in the near future.

Reply:
From offline discussion with Wayne, if there is no "spiffe://" prefix, assign the uri to user and return true. This helps to keep consistent behavior for cases where uri does not begin with "spiffe://".